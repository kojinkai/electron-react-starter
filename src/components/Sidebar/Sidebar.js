import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './Sidebar.css';

export class Sidebar extends Component {
  render() {
    return (
      <nav className="sidebar">
        <Link to="/">Dashboard</Link>
      </nav>
    );
  }
}

export default Sidebar;
