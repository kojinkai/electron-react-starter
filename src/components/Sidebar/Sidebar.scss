@import 'styles/grid';
@import 'styles/colors';

.sidebar {
  background: $core;
  color: $white;
  display: flex;
  flex: 1;
  flex-direction: column;
  min-width: 220px;

  &__item {
    padding: $gutter;
  }
}
