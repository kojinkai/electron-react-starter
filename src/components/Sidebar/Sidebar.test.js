import React from 'react';
import { shallow } from 'enzyme';
import Sidebar from './Sidebar';

it('renders according to the snapshot', () => {
  const wrapper = shallow(<Sidebar />);
  expect(wrapper).toMatchSnapshot();
});
