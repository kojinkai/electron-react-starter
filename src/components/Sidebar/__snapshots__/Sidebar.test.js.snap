// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders according to the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="sidebar__item"
          replace={false}
          to="/"
>
          Dashboard
</Link>,
        <Link
          className="sidebar__item"
          replace={false}
          to="/settings"
>
          Settings
</Link>,
      ],
      "className": "sidebar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Dashboard",
          "className": "sidebar__item",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Dashboard",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Settings",
          "className": "sidebar__item",
          "replace": false,
          "to": "/settings",
        },
        "ref": null,
        "rendered": "Settings",
        "type": [Function],
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="sidebar__item"
            replace={false}
            to="/"
>
            Dashboard
</Link>,
          <Link
            className="sidebar__item"
            replace={false}
            to="/settings"
>
            Settings
</Link>,
        ],
        "className": "sidebar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Dashboard",
            "className": "sidebar__item",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Dashboard",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Settings",
            "className": "sidebar__item",
            "replace": false,
            "to": "/settings",
          },
          "ref": null,
          "rendered": "Settings",
          "type": [Function],
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
